common/scala/src/main/scala/whisk/core/entity/EntityPath.scala

```diff
@@ -155,8 +156,20 @@ protected[core] object EntityName {
 }
 
 /**
- * An EntityQName (qualified name) is a pair of an EntityPath and and EntityName.
+ * A FullyQualifiedEntityName (qualified name) is a triple consisting of
+ * - EntityPath: the namespace and package where the entity is located
+ * - EntityName: the name of the entity
+ * - Version: a unique version for the resource
  *
- * Note: by convention, a "fully qualified name" would additionally include a version
+ * The version is not a SemVer (yet) because it semantic versioning of entities
+ * is not enforced. Instead this will be a context specific version identifier.
  */
-protected[core] case class EntityQName(path:EntityPath, name:EntityName)
+protected[core] case class FullyQualifiedEntityName(path: EntityPath, name: EntityName, version: Option[String]) {
+    override def toString = {
+        s"$path${EntityPath.PATHSEP}$name${version.map { v => s"@$v" }.getOrElse("")}"
```

Comment:
String interpolation actually looks very clunky in this case.

`path + EntityPath.PATHSEP + name + version.map("@" + _).getOrElse("")` is a little easier on the eye i think?